### PR TITLE
Aidan dev

### PR DESCRIPTION
**Pull request by aidanhchau**

### CODE REVIEW

**Comment by aidanhchau** [comment]
automated iteration, model training, and logging through different hyperparameters (loss functions and levels) using itertools
